### DISCUSSION TITLE
Compatibility with 3D kspace

### DISCUSSION THREAD

**Discussion by tamas-borbath-orbem** [facebookresearch/fastMRI#208] [Q&A]
Hi,
I am considering to use the fastMRI model for an undersampled 3 dimensional k-space (two phase encoding directions) and not 2 dimensional slices of k-space (one slice encoding and one phase encoding), as it is currently in all the datasets.
Does anyone have experience with this?
Thank you very much!
All the best,
Tamas

**Comment by mmuckley** [comment]
Hello @tamas-borbath-orbem, our current code base in open source would not work with this. You would probably mostly need to rewrite how [subsampling] is done and then write a new [DataTransform](https://github.com/facebookresearch/fastMRI/blob/main/fastmri/data/transforms.py) for your case.

I don't know of any projects that have done this but there might be some out there.

**Reply by tamas-borbath-orbem** [reply]
Thank you very much for your answer! Much appreciated!
In case I will solve it, will contribute with a pull request

**Reply by mmuckley** [reply]
@tamas-borbath-orbem just one thing to note: in terms of including your code in the repo: we would merge it only if it were in scope (i.e., works with the fastMRI data). Otherwise, you would have your implementation in your own repository. That being said if you wanted to advertise your 3D models here on the Discussions page I think that would be fine :).

**Reply by tamas-borbath-orbem** [reply]
Sure! :)
